hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -131,12 +145,223 @@ public void assignDistributionSetMultipleTimes() {
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 6),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 2),
+            @Expect(type = TargetUpdatedEvent.class, count = 3), @Expect(type = TargetPollEvent.class, count = 1) })
+    public void finishActionInMultiAssignMode() {
+        setMultiAssignmentsEnabled(true);
+        final String controllerId = TARGET_PREFIX + "finishActionInMultiAssignMode";
+        registerAndAssertTargetWithExistingTenant(controllerId);
+
+        final long actionId1 = assignNewDsToTarget(controllerId);
+        final long actionId2 = assignNewDsToTarget(controllerId);
+        final Entry<Long, EventTopic> action2Install = new SimpleEntry<>(actionId2, EventTopic.DOWNLOAD_AND_INSTALL);
+        waitUntilEventMessagesAreSent(2);
+
+        updateActionViaDmfClient(controllerId, actionId1, DmfActionStatus.FINISHED);
+        waitUntilEventMessagesAreSent(4);
+        assertRequestAttributesUpdateMessage(controllerId);
+        assertLatestMultiActionMessage(controllerId, Arrays.asList(action2Install));
+    }
+
+    @Test
+    @Description("If multi assignment is enabled assigning a DS multiple times creates a new action every time.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 1),
+            @Expect(type = DeploymentEvent.class, count = 2),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 0),
+            @Expect(type = CancelTargetAssignmentEvent.class, count = 0),
+            @Expect(type = ActionCreatedEvent.class, count = 2), @Expect(type = ActionUpdatedEvent.class, count = 0),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 3),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 1),
+            @Expect(type = TargetUpdatedEvent.class, count = 2), @Expect(type = TargetPollEvent.class, count = 1) })
+    public void assignDsMultipleTimesInMultiAssignMode() {
+        setMultiAssignmentsEnabled(true);
+        final String controllerId = TARGET_PREFIX + "assignDsMultipleTimesInMultiAssignMode";
+        registerAndAssertTargetWithExistingTenant(controllerId);
+        final DistributionSet ds = testdataFactory.createDistributionSet(UUID.randomUUID().toString());
+
+        final Long actionId1 = assignDistributionSet(ds.getId(), controllerId).getActionIds().get(0);
+        waitUntilEventMessagesAreSent(1);
+        final Long actionId2 = assignDistributionSet(ds.getId(), controllerId).getActionIds().get(0);
+        waitUntilEventMessagesAreSent(2);
+        
+        final Entry<Long, EventTopic> action1Install = new SimpleEntry<>(actionId1, EventTopic.DOWNLOAD_AND_INSTALL);
+        final Entry<Long, EventTopic> action2Install = new SimpleEntry<>(actionId2, EventTopic.DOWNLOAD_AND_INSTALL);
+        assertLatestMultiActionMessage(controllerId, Arrays.asList(action1Install, action2Install));
```

Comment:
re-visit assertLatestMultiActionMessage which does not need to enforce the order...